### PR TITLE
Cache the paper key in paperprovision command

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

**Comment by jzila** [comment]
Need this to validate paper key device provisioning in Docker tests. Anyone can give me a thumb, but I tagged all who might be interested.

**Comment by zanderz** [comment]
I'll have to defer to the others

**Comment by maxtaco** [comment]
I'm going to defer to @patrickxb, I know he recently changed this.

File: go/engine/paperprovision.go [hunk 1]
```diff
@@ -117,6 +117,11 @@ func (e *PaperProvisionEngine) Run(ctx *Context) (err error) {
 	if err := tx.Commit(); err != nil {
 		return err
 	}
+	if err := e.G().LoginState().Account(func(a *libkb.Account) {
+		a.SetUnlockedPaperKey(kp.sigKey, kp.encKey)
```
Comment (patrickxb) [inline go/engine/paperprovision.go]: Just put this one line (121) after line 169 instead.

Comment (patrickxb) [inline go/engine/paperprovision.go]: here: https://github.com/keybase/client/blob/82d7884f6f1af3f25e61a1e2798660482788d2a0/go/engine/paperprovision.go#L169

File: go/engine/paperprovision.go [hunk 2]
```diff
@@ -167,6 +162,11 @@ func (e *PaperProvisionEngine) paper(ctx *Context, kp *keypair) error {
 		if err := e.makeDeviceKeysWithSigner(ctx, kp.sigKey); err != nil {
 			return err
 		}
+		if err := e.G().LoginState().Account(func(a *libkb.Account) {
+			a.SetUnlockedPaperKey(kp.sigKey, kp.encKey)
+		}, "UnlockedPaperKey"); err != nil {
+			return err
+		}
```
Comment (patrickxb) [inline go/engine/paperprovision.go]: Hmm.  I was hoping this was going to be one line:

    lctx.SetUnlockedPaperKey(....)

But that method isn't currently in LoginContext.  The above LoginState() block isn't going to work as this code is already in a serialized LoginState process.

Options:

1. Add `SetUnlockedPaperKey` to LoginContext interface and use the line above.
2. Put this block back where you had it.

I'd prefer option 1.

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**